lib/buffer.js

```diff
@@ -658,6 +658,87 @@ Buffer.prototype.includes = function includes(val, byteOffset, encoding) {
   return this.indexOf(val, byteOffset, encoding) !== -1;
 };
 
+Buffer.prototype.indexes = function indexes(match, limit) {
+  if (!Buffer.isBuffer(match))
+    throw new TypeError('argument must be a Buffer');
+  const indexes = [];
+  let i = -1;
+
+  let len = 0
+  while ((i = this.indexOf(match, i + 1)) != -1) {
+    indexes.push(i);
+    len++;
+    if (len >= limit)
+      break;
+  }
+  return indexes;
+};
+
+Buffer.prototype.split = function split(sep, encoding, limit) {
+  if (!limit && typeof encoding === 'number')
+    limit = encoding;
+  if (sep)
+    sep = Buffer.from(sep, encoding); // allows all constructor APIs
+
+  const res = [];
+  // false case: return Array of 1 Buffer
+  const sepLen = sep ? sep.byteLength : undefined;
+  const indexes = sep ? this.indexes(sep, limit) : [];
+
+  if (sepLen > 0) {
+    let cursor = 0;
+    const len = indexes.length;
+    // push from index to index; catch last case after loop
+    for (let i = 0; i < len; i++) {
+      res.push(this.slice(cursor, indexes[i]));
+      cursor += indexes[i] - cursor + sepLen;
```

Comment:
I think the whole point of this is to have compatibility with `String.prototype.split`, right? I think you have to handle the case where the `sep` strings overlap differently, then:
```js
> 'aaabaaa'.split('aa')
[ '', 'ab', 'a' ]
> Buffer.from('aaabaaa').split(Buffer.from('aa'))
[ <Buffer >, <Buffer >, <Buffer 62>, <Buffer >, <Buffer > ]
```